astropy/table/tests/test_table.py

```diff
@@ -2331,7 +2331,10 @@ def test_units(self):
             table.Table.from_pandas(df, units=[u.m, u.s])
 
         # test warning is raised if additional columns in units dict
-        with pytest.warns(UserWarning) as record:
+        with pytest.warns(
+            UserWarning,
+            match=r"^`units` contains additional columns: {'y'}$",
+        ) as record:
             table.Table.from_pandas(df, units={"x": u.m, "t": u.s, "y": u.m})
         assert len(record) == 1
         assert "{'y'}" in record[0].message.args[0]
```

Comment:
```suggestion
```

Reply:
nice catch, I think the other assert can be dropped too. Thanks !